From 2ab9057a452773eed29e715ff7a3ec2a778f6220 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Fri, 29 Mar 2024 14:53:08 +0100
Subject: [PATCH 1059/1141] arm64: dts: st: update hdmi node in stm32mp257f-ev1

The version of hdmi bridge supported by the eval board
is no longer adv7535 but adv7533.

Change-Id: I493f7ca4cdd61824e2eed232ca1d552ebc8abb39
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/370264
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index 18af06ba5850..7c27e597e3c9 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -92,7 +92,7 @@ hdmi: connector {
 
 		port {
 			hdmi_connector_in: endpoint {
-				remote-endpoint = <&adv7535_out>;
+				remote-endpoint = <&adv753x_out>;
 			};
 		};
 	};
@@ -224,7 +224,7 @@ dsi_in: endpoint {
 		port@1 {
 			reg = <1>;
 			dsi_out1: endpoint {
-				remote-endpoint = <&adv7535_in>;
+				remote-endpoint = <&adv753x_in>;
 			};
 		};
 	};
@@ -328,8 +328,12 @@ imx335_ep: endpoint {
 		};
 	};
 
-	adv7535: hdmi@3d {
-		compatible = "adi,adv7535";
+	adv753x: hdmi@3d {
+		/*
+		 * With MB1232 board, use "adi,adv7533" (1080p30)
+		 * With MB1752 board, use "adi,adv7535" (1080p60)
+		 */
+		compatible = "adi,adv7533";
 		reg = <0x3d>, <0x3c>, <0x3f>, <0x38>;
 		reg-names = "main", "cec", "edid", "packet";
 		status = "disabled";
@@ -348,21 +352,21 @@ ports {
 
 			port@0 {
 				reg = <0>;
-				adv7535_in: endpoint {
+				adv753x_in: endpoint {
 					remote-endpoint = <&dsi_out1>;
 				};
 			};
 
 			port@1 {
 				reg = <1>;
-				adv7535_out: endpoint {
+				adv753x_out: endpoint {
 					remote-endpoint = <&hdmi_connector_in>;
 				};
 			};
 
 			port@2 {
 				reg = <2>;
-				adv7535_tx_endpoint: endpoint {
+				adv753x_tx_endpoint: endpoint {
 					remote-endpoint = <&i2s2_endpoint>;
 				};
 			};
@@ -400,7 +404,7 @@ &i2s2 {
 
 	i2s2_port: port {
 		i2s2_endpoint: endpoint {
-			remote-endpoint = <&adv7535_tx_endpoint>;
+			remote-endpoint = <&adv753x_tx_endpoint>;
 			format = "i2s";
 			mclk-fs = <256>;
 		};
-- 
2.39.2

